Added handing for simpler error message for Unknown Dockerfile instru…

ipbabble opened the pull request:
…ctions.

Signed-off-by: William Henry <whenry@redhat.com>

ipbabble commented:
Not retest this please

ipbabble commented:
Bot restart this please

ipbabble commented:
Bot retest this please

ripcurld0 commented on imagebuildah/build.go:
```diff
@@ -389,12 +389,20 @@ func (b *Executor) Run(run imagebuilder.Run, config docker.Config) error {
 	}
-	logrus.Errorf("+(UNIMPLEMENTED KEYWORD %q?) %#v", step.Command, step)
-	return errors.Errorf("Unrecognized instruction %q: %#v", step.Command, step)
+	if logrus.GetLevel() == logrus.ErrorLevel {
```
I would rather to use `switch` here:

```
switch logrus.GetLevel() {
case logrus.ErrorLevel:
  ...
case logrus.DebugLevel:
  ...
default:
   return errors.Errorf(err)
}
```

rhatdan replied:
Yes use switch statement.

ripcurld0 commented on imagebuildah/build.go:
```diff
@@ -389,12 +389,20 @@ func (b *Executor) Run(run imagebuilder.Run, config docker.Config) error {
-	logrus.Errorf("+(UNIMPLEMENTED KEYWORD %q?) %#v", step.Command, step)
-	return errors.Errorf("Unrecognized instruction %q: %#v", step.Command, step)
+	if logrus.GetLevel() == logrus.ErrorLevel {
+		fmt.Println(err_str)
```
I am curious. why are you using `fmt.Println` and not `logrus.Debug` f.e.?

rhatdan replied:
Definitely shold not be fmt.Println(err_str).

Either logrus.Debug or logrus.Error.

ipbabble replied:
@rhatdan You see it returns a errors.ErrorF at the end? After it falls out of if else.
 Let me see what logrus.Error does.

ipbabble replied:
Ok so I tested logrus.ErrorF   (no .Errors) and I will use that instead of the Println. Thanks for the heads up on that.

rhatdan commented on imagebuildah/build.go:
```diff
@@ -389,12 +389,20 @@ func (b *Executor) Run(run imagebuilder.Run, config docker.Config) error {
 // imagebuilder parser didn't understand.
 func (b *Executor) UnrecognizedInstruction(step *imagebuilder.Step) error {
+	err_str := fmt.Sprintf("Build error: Unknown instruction: %q ", step.Command)
+	err := fmt.Sprintf(err_str+"%#v", step)
```
These lines should be combined.

err := fmt.Sprintf("Build error: Unknown instruction: %q  %#v", ste.Command, step)

ipbabble replied:
They are different. One is giving the full step info. The other is trying to simplify for less verbose output.

rhatdan replied:
Ok you are using them in different output commands. Never mind.

ipbabble commented:
I’m using Println just like we do for outputing STEP output. I.e. if I’m a
user NOT in debug mode I’d like to have a simple output that tells me the
problem instead of having to wade through DebugF output.  So I use a
printlin

Having said that, the only way to exit this func in error and stop the
build run, is to return an error. So we are going to get some spew. I tried
to put out a clean line and then limit the spew.

If you look at Docker in non debug, all you get is a simple Unknown
Instruction error line.

Thanks for asking the question and forcing me to clarify my point.

William

On Thu, Feb 8, 2018 at 6:18 AM Daniel J Walsh <notifications@github.com>
wrote:

> *@rhatdan* commented on this pull request.
> ------------------------------
>
> In imagebuildah/build.go
> <https://github.com/projectatomic/buildah/pull/457#discussion_r166929154>:
>
> > @@ -389,12 +389,20 @@ func (b *Executor) Run(run imagebuilder.Run, config docker.Config) error {
>  // UnrecognizedInstruction is called when we encounter an instruction that the
>  // imagebuilder parser didn't understand.
>  func (b *Executor) UnrecognizedInstruction(step *imagebuilder.Step) error {
> +	err_str := fmt.Sprintf("Build error: Unknown instruction: %q ", step.Command)
> +	err := fmt.Sprintf(err_str+"%#v", step)
>
> These lines should be combined.
>
> err := fmt.Sprintf("Build error: Unknown instruction: %q %#v",
> ste.Command, step)
> ------------------------------
>
> In imagebuildah/build.go
> <https://github.com/projectatomic/buildah/pull/457#discussion_r166929309>:
>
> >  		return nil
>  	}
> -	logrus.Errorf("+(UNIMPLEMENTED KEYWORD %q?) %#v", step.Command, step)
> -	return errors.Errorf("Unrecognized instruction %q: %#v", step.Command, step)
> +	if logrus.GetLevel() == logrus.ErrorLevel {
>
> Yes use switch statement.
> ------------------------------
>
> In imagebuildah/build.go
> <https://github.com/projectatomic/buildah/pull/457#discussion_r166929587>:
>
> >  		return nil
>  	}
> -	logrus.Errorf("+(UNIMPLEMENTED KEYWORD %q?) %#v", step.Command, step)
> -	return errors.Errorf("Unrecognized instruction %q: %#v", step.Command, step)
> +	if logrus.GetLevel() == logrus.ErrorLevel {
> +		fmt.Println(err_str)
>
> Definitely shold not be fmt.Println(err_str).
>
> Either logrus.Debug or logrus.Error.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/projectatomic/buildah/pull/457#pullrequestreview-95067324>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABPpgVzpjQqOKNUv9mbLtSo9RMK1Tsuoks5tSvQKgaJpZM4R9di5>
> .
>
-- 
Sent from Gmail Mobile on iPhone

rhatdan commented:
logrus.Errorf should hopefully go to stderr, while fmt.Println goes to stdout.

ipbabble commented:
I've switched to switch and I've switched to  logrus.ErrorF().  Should be good.
Also added a default "UNKNOWN LOGLEVEL" so we can see if we need to add special code later.

ipbabble commented:
@rhatdan @TomSweeneyRedHat changes made and waiting on your blessing.

TomSweeneyRedHat commented:
shouldn't hit the close button (again).

ipbabble commented:
Ack

On Thu, Feb 8, 2018 at 12:42 PM Tom Sweeney <notifications@github.com>
wrote:

> Changes LGTM, but you've a gofmt error. Likely a misplaced space.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/projectatomic/buildah/pull/457#issuecomment-364222328>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABPpgQy8GcJHnL_t03mVdLJLhbNpu2Wnks5tS0svgaJpZM4R9di5>
> .
>
-- 
Sent from Gmail Mobile on iPhone

ipbabble commented:
@TomSweeneyRedHat gofmt done

rhatdan commented:
@rh-atomic-bot r+
